compiler/quilt/tools/build.py

```diff
@@ -63,7 +63,7 @@ def _path_hash(path, transform, kwargs):
     return digest_string(srcinfo)
 
 def _is_internal_node(node):
-    is_leaf = not node or node.get(RESERVED['file'])
+    is_leaf = not node or isinstance(node.get(RESERVED['file']), str)
```

Comment:
So what exactly is this fixing? The unit test is passing with or without this change.

Reply:
Currently without this fix, any 'quilt/compiler/quilt/test/data/file.*' will fail two tests in `test_build.py`, because node parser will think that it reached leaf but in reality it's not.

`node` variable indicates kinda current depth level of the build file, and leaf of the build config might be path to the file, for example, `file: data/foo.csv`. Also, before reaching leaf, node contains all parsed files, for example,`node` for `build.yml` right before going one level lower will look similar to this:
`node = {'README': {'file': 'README.md'}, 'bad': {'file': 'bad.csv'}, 'csv': {'file': 'csv.txt'}, ...}`
But what if there will be a file, e.g., called 'file.csv', then node will look like this:
`node = {'README': {'file': 'README.md'}, 'bad': {'file': 'bad.csv'}, 'csv': {'file': 'csv.txt'}, 'file': {'file': 'file.csv'}, ...}`
In this case, node parser will expose it as leaf since there is a key `file`, but it's not actually a leaf. Thus, dict will pass as path to the file, which will fail on `join()` eventually. Actually, exactly this error was occurred in the issue ticket #314.

I am sorry for many details, I just wanted to be as clear as possible.